reV/bespoke/bespoke.py

```diff
@@ -521,14 +568,78 @@ def plant_optimizer(self):
-                                             self.include_mask,
-                                             self.pixel_side_length,
-                                             self.min_spacing)
+            self._plant_optm = PlaceTurbines(
+                self.wind_plant_pd,
+                self.objective_function,
+                self.capital_cost_function,
+                self.fixed_operating_cost_function,
+                self.variable_operating_cost_function,
+                self.include_mask,
+                self.pixel_side_length,
+                self.min_spacing)
+
         return self._plant_optm
 
+    def recalc_lcoe(self):
+        """Recalculate the multi-year mean LCOE based on the multi-year mean
+        annual energy production (AEP)"""
+
+        if 'lcoe_fcr-means' in self.outputs:
+            lcoe_kwargs = self.get_lcoe_kwargs()
+
+            logger.debug('Recalulating multi-year mean LCOE using '
+                         'multi-year mean AEP.')
+
+            fcr = lcoe_kwargs['fixed_charge_rate']
+            cap_cost = lcoe_kwargs['capital_cost']
+            foc = lcoe_kwargs['fixed_operating_cost']
+            voc = lcoe_kwargs['variable_operating_cost']
+            aep = self.outputs['annual_energy-means']
+
+            my_mean_lcoe = lcoe_fcr(fcr, cap_cost, foc, aep, voc)
+
+            self._outputs['lcoe_fcr-means'] = my_mean_lcoe
+            self._meta['mean_lcoe'] = my_mean_lcoe
+
+    def get_lcoe_kwargs(self):
+        """Get a namespace of arguments for calculating LCOE based on the
+        bespoke optimized wind plant capacity
+
+        Returns
+        -------
+        lcoe_kwargs : dict
+            kwargs for the SAM lcoe model. These are based on the original
+            sam_sys_inputs, normalized to the original system_capacity, and
+            updated based on the bespoke optimized system_capacity, includes
+            fixed_charge_rate, system_capacity (kW), capital_cost ($),
+            fixed_operating_cos ($), variable_operating_cost ($/kWh)
+        """
+
+        if 'system_capacity' not in self.outputs:
+            msg = ('Could not find system_capacity in the outputs, need to '
+                   'run_plant_optimization() to get the optimized '
+                   'system_capacity before calculating LCOE!')
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        fcr = self.original_sam_sys_inputs['fixed_charge_rate']
+        sys_cap = self.plant_optimizer.capacity
+        cap_cost = self.plant_optimizer.capital_cost
+        foc = self.plant_optimizer.fixed_operating_cost
+        voc = self.plant_optimizer.variable_operating_cost
+
+        lcoe_kwargs = {}
+        lcoe_kwargs['fixed_charge_rate'] = fcr
+        lcoe_kwargs['system_capacity'] = sys_cap
+        lcoe_kwargs['capital_cost'] = cap_cost
+        lcoe_kwargs['fixed_operating_cost'] = foc
+        lcoe_kwargs['variable_operating_cost'] = voc
```

Comment:
Any reason not to just build a dictionary right away?
```suggestion
        lcoe_kwargs['variable_operating_cost'] = voc
```
```suggestion
        lcoe_kwargs = {
            'fixed_charge_rate': self.original_sam_sys_inputs['fixed_charge_rate']
            'system_capacity': self.plant_optimizer.capacity
            'capital_cost': self.plant_optimizer.capital_cost
            'fixed_operating_cost': self.plant_optimizer.fixed_operating_cost
            'variable_operating_cost': self.plant_optimizer.variable_operating_cost
        }
```